internal/middleware/access.go

```diff
@@ -0,0 +1,25 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/dilyara4949/flight-booking-api/internal/handler/response"
+	"github.com/gin-gonic/gin"
+)
+
+const userRole = "user"
+
+func AccessCheck(role string, param string) gin.HandlerFunc {
```

Comment:
I don't think, this middleware would work as intended. 
We need to check for elevated privileges. here, we only check if user role is "user".
It would be better to get claims from the context and check them against required role( or roles) from the function parameter. 

E.g. user claims from JWT state that his role is user, we have AccessCheck middleware:
```go
func AccessCheck(allowedRoles ...string) gin.HandlerFunc {
```
Although, this would only work if you want to limit access to the endpoint itself, in case, that you need to check authorization for the particular handler it would be probably better to have a AccessCheck method on the service, smth like:

```go
type HasID interface { 
  // your requests need to implement this interface, if you want to check them with this finction
  // alternatively you can use reflection, but that would make it more obscure
  func GetID() string
}

func AccessCheck(ctx context.Context, req HasID, id string, roles ...string) bool {
 // here we check if user have role or his ID matches that of the context or his role in the allowed list. 
 // If list is empty, then we check only ID 
}
```
Then , when you need to check access you need just perform the check in your handler body:
```go
if !AccessCheck(c, req, id, requiredRole) {
  c.AbortWithStatusJSON(http.StatusForbidden, response.Error{Error: "access denied"})
}

```